llvm/test/CodeGen/RISCV/cheri/ptradd-dagcombine-infinite.ll

```diff
@@ -0,0 +1,20 @@
+; This IR previously triggered an infinite DAGCombiner loop inside DAGCombiner::visitPTRADD().
+; Check that this no longer happens and that we can fold the two GEPs into a single ISD::PTRADD.
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: %riscv64_cheri_purecap_llc -verify-machineinstrs -O0 < %s | FileCheck %s
+
+%struct.ED_ = type { i64, { i64, i64 }, i8 }
+
+define internal %struct.ED_ addrspace(200)* @bsdf_is_edge(i64 %arg1, %struct.ED_ addrspace(200)* %arg2) addrspace(200) {
+; CHECK-LABEL: bsdf_is_edge:
+; CHECK:       # %bb.0: # %if.then81
+; CHECK-NEXT:    cincoffset ca0, ca1, -32
+; CHECK-NEXT:    cret
+if.then81:
+  %mul83 = mul nsw i64 0, %arg1
+  %add.ptr85 = getelementptr inbounds %struct.ED_, %struct.ED_ addrspace(200)* %arg2, i64 %mul83
+  %add.ptr86 = getelementptr inbounds %struct.ED_, %struct.ED_ addrspace(200)* %add.ptr85, i64 -1
+  ret %struct.ED_ addrspace(200)* %add.ptr86
+}
+
+attributes #0 = { nofree nosync nounwind readnone speculatable willreturn }
```

Comment:
Should be able to clean up this IR (struct.ED_ -> struct, drop the attributes (other than nounwind which belongs on the function itself) and explicit function AS, drop the BB label, rename all the variables to be %a/%b for arguments and %0/%1/%2 for locals)